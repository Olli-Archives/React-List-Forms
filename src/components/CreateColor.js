import React from 'react';
import PropTypes from 'prop-types';

export default function CreateColor({ color }) {

  const { name, hex, rgb } = color;

  return (
    <dl>
      <dt>name</dt>
      <dd>{name}</dd>
      <dt>hex:{name}</dt>
      <dd>{hex}</dd>
      <dt>RGB</dt>
      <dd>
        <p>red:{rgb.red}</p>
        <p>green:{rgb.green}</p>
        <p>blue:{rgb.blue}</p>
      </dd>
    </dl>
  );
}

CreateColor.propTypes = {
  color:PropTypes.object.isRequired
};
