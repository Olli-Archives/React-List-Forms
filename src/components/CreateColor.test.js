import React from 'react';
import { shallow } from 'enzyme';
import CreateColor from './CreateColor';

describe('CreateColor component', ()=>{
  it('it creates a color with an object property', ()=>{
    const color = {
      name:'blue',
      rgb:{
        red:2,
        green:9,
        blue:44
      }
    };
    const wrapper = shallow(<CreateColor color={color}/>);
    expect(wrapper).toMatchSnapshot();
  });
})
;
