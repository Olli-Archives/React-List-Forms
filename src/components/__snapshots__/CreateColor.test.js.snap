// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateColor component it creates a color with an object property 1`] = `
<dl>
  <dt>
    name
  </dt>
  <dd>
    blue
  </dd>
  <dt>
    hex:
    blue
  </dt>
  <dd />
  <dt>
    RGB
  </dt>
  <dd>
    <p>
      red:
      2
    </p>
    <p>
      green:
      9
    </p>
    <p>
      blue:
      44
    </p>
  </dd>
</dl>
`;
